specify exact ndk version

If there is an ndk installed already it most likely can't be specified with this syntax in .axproj
`ndk=r26`

add ndkVersion, such that an exact version can be specified
`ndkVersion=26.1.10909125`

Also useful, an option to use latest already intalled ndk if version greater than min required, maybe
ndk=latest
or some other key

not working at all, i think. trying to build with axmol build:

```
PS C:\Users\User\Desktop\projects\axprojects\MultiFocus> type .axproj
package_name=com.iandyhd.multifocus
engine_version=2.4.0-2808a8f
project_type=cpp
ndk=r26+
PS C:\Users\User\Desktop\projects\axprojects\MultiFocus> axmol build -p android -a arm64
axmol: -p android -a arm64 -d: C:\Users\User\Desktop\projects\axprojects\MultiFocus
1kiss: PowerShell 7.4.7 on Microsoft Windows 10.0.22631.0
1kiss:
Name                           Value
----                           -----
prefix                         C:\Users\User\Desktop\projects\axprojects\axmol\tools\external
scope                          local
aab                            False
t
a                              arm64
j                              24
cc
dm                             False
xc                             {}
u                              False
d                              C:\Users\User\Desktop\projects\axprojects\MultiFocus
O                              -1
xb                             {}
i                              False
dll                            False
minsdk
sdk
p                              android
xt                             proj.android/gradlew


1kiss: proj_dir=C:\Users\User\Desktop\projects\axprojects\MultiFocus, install_prefix=C:\Users\User\Desktop\projects\axprojects\axmol\tools\external
1kiss: Using axslcc: C:\Users\User\Desktop\projects\axprojects\axmol\tools\external\axslcc\axslcc.exe, version: 1.9.6
C:\Users\User\Desktop\projects\axprojects\axmol\tools\external\axslcc\axslcc.exe
1kiss: Using cmake: C:\Program Files\CMake\bin\cmake.exe, version: 3.30.5
1kiss: info: the cmd follow symlink C:\Program Files\CMake\bin\ninja.exe ==> C:\Users\User\AppData\Local\Programs\Python\Python311\Scripts\ninja.exe
1kiss: Using ninja: C:\Program Files\CMake\bin\ninja.exe, version: 1.11.1
1kiss: info: the cmd follow symlink C:\Program Files\CMake\bin\ninja.exe ==> C:\Users\User\AppData\Local\Programs\Python\Python311\Scripts\ninja.exe
1kiss: Using jdk: C:\Users\User\Desktop\projects\axprojects\axmol\tools\external\jdk\bin\javac.exe, version: 17.0.13
1kiss: Looking require r23d in C:\Users\User\AppData\Local\Android\Sdk
1kiss: Using cmdline-tools: C:\Users\User\AppData\Local\Android\Sdk\cmdline-tools\12.0\bin\sdkmanager.bat, version: 12.0
1kiss: Not found suitable android ndk, installing from ci.android.com ...
1kiss: Downloading https://androidbuildinternal.googleapis.com/android/internal/build/v3/builds/12186248/win64/attempts/latest/artifacts/android-ndk-12186248-windows-x86_64.zip/url to C:\Users\User\Desktop\projects\axprojects\axmol\cache\android-ndk-12186248-windows-x86_64.zip ...
PS C:\Users\User\Desktop\projects\axprojects\MultiFocus> ^C
PS C:\Users\User\Desktop\projects\axprojects\MultiFocus>
```

android studio says:
Caused by: java.lang.Exception: No installed ndk found, required 26.-54.*

--- a/1k/1kiss.ps1
+++ b/1k/1kiss.ps1
@@ -231,7 +231,9 @@ $manifest = @{
     python       = '3.8.0+';
     jdk          = '17.0.10+'; # jdk17+ works for android cmdlinetools 7.0+
     emsdk        = '3.1.53+';
-    'cmdline-tools' = '12.0'; # android cmdlinetools
+    cmdlinetools = '12.0'; # android cmdline-tools
+    buildtools   = '34.0.0'; # android build-tools
+    target_sdk   = '35'      # android platforms
 }
 
 # the default generator requires explicit specified: osx, ios, android, wasm
@@ -252,12 +254,6 @@ $cmdlinetools_rev = '11076708' # 12.0
 
 $ndk_r23d_rev = '12186248'
 # $ndk_r25d_rev = '12161346'
-
-$android_sdk_tools = @{
-    'cmdline-tools' = '12.0'
-    'build-tools' = '34.0.0'
-    'platforms'   = '35'
-}
 
 # eva: evaluated_args
 $options = @{
@@ -474,6 +470,14 @@ $Global:is_msvc = $TOOLCHAIN_NAME -eq 'msvc'
 $manifest_file = Join-Path $myRoot 'manifest.ps1'
 if ($1k.isfile($manifest_file)) {
     . $manifest_file
+}
+
+if($1k.isfile($Global:__1k_user_profile)) {
+    $1k.println("Loading user build profile: $__1k_user_profile")
+    $user_profile = ConvertFrom-Props (Get-Content $__1k_user_profile)
+    foreach($entry in $user_profile.GetEnumerator()) {
+        $manifest[$entry.Key] = $entry.Value
+    }
 }
 
 $install_prefix = if ($options.prefix) { $options.prefix } else { Join-Path $HOME '.1kiss' }
@@ -1306,11 +1310,11 @@ function setup_android_sdk() {
     $sdk_comps = @()
 
     ### cmdline-tools ###
-    $cmdlinetools_ver = $($android_sdk_tools['cmdline-tools'])
+    $cmdlinetools_ver = $manifest['cmdlinetools']
     $sdkmanager_prog, $sdkmanager_ver = $null, $null
     $cmdlinetools_prefix = Join-Path $sdk_root "cmdline-tools"
     $cmdlinetools_bin = Join-Path $cmdlinetools_prefix "$cmdlinetools_ver/bin"
-    $sdkmanager_prog, $sdkmanager_ver = (find_prog -name 'cmdline-tools' -cmd 'sdkmanager' -path $cmdlinetools_bin -params "--version", "--sdk_root=$sdk_root")
+    $sdkmanager_prog, $sdkmanager_ver = (find_prog -name 'cmdlinetools' -cmd 'sdkmanager' -path $cmdlinetools_bin -params "--version", "--sdk_root=$sdk_root")
     if (!$sdkmanager_prog) {
         $suffix = $('win', 'linux', 'mac').Get($HOST_OS)
         if (!$sdkmanager_prog) {
@@ -1319,7 +1323,7 @@ function setup_android_sdk() {
             $cmdlinetools_pkg_name = "commandlinetools-$suffix-$($cmdlinetools_rev)_latest.zip"
             $cmdlinetools_url = "https://dl.google.com/android/repository/$cmdlinetools_pkg_name"
             fetch_pkg $cmdlinetools_url -o $cmdlinetools_pkg_name -exrep "cmdline-tools=$cmdlinetools_ver" -prefix $cmdlinetools_prefix
-            $sdkmanager_prog, $_ = (find_prog -name 'cmdline-tools' -cmd 'sdkmanager' -path $cmdlinetools_bin -params "--version", "--sdk_root=$sdk_root" -silent $True)
+            $sdkmanager_prog, $_ = (find_prog -name 'cmdlinetools' -cmd 'sdkmanager' -path $cmdlinetools_bin -params "--version", "--sdk_root=$sdk_root" -silent $True)
             if (!$sdkmanager_prog) {
                 throw "Install cmdlinetools version: $sdkmanager_ver fail"
             }
@@ -1353,7 +1357,7 @@ function setup_android_sdk() {
             if (!$1k.isdir($ndk_root)) { throw "Install android-ndk-r23d fail, please try again" }
         }
         else {
-            $1k.println("Not found suitable android ndk, installing by sdkmanager ...")
+            $1k.println("Not found suitable android ndk, installing ndk-$ndk_ver by sdkmanager ...")
 	    
             $matchInfos = (exec_prog -prog $sdkmanager_prog -params "--sdk_root=$sdk_root", '--list' | Select-String 'ndk;')
             if ($null -ne $matchInfos -and $matchInfos.Count -gt 0) {
@@ -1382,7 +1386,7 @@ function setup_android_sdk() {
     }
 
     if (!$ndkOnly) {
-        $sdk_comps_list = 'platform-tools', "platforms/android-$($android_sdk_tools['platforms'])", "build-tools/$($android_sdk_tools['build-tools'])"
+        $sdk_comps_list = 'platform-tools', "platforms/android-$($manifest['target_sdk'])", "build-tools/$($manifest['buildtools'])"
         foreach ($comp in $sdk_comps_list) {
             if (!$1k.isfile("$sdk_root/$comp/source.properties") -or $updateAdt) {
                 $sdk_comps += $comp.Replace('/', ';')

--- a/1k/build.profiles
+++ b/1k/build.profiles
@@ -37,7 +37,7 @@ llvm=17.0.6~19.1.6+
 jdk=17.0.10~17.0.13+
 
 # The android cmdline-tools aka sdkmanager, as stable as possible
-cmdline-tools=12.0
+cmdlinetools=12.0
 
 # The android ndk version
 # as stable as possible @setup.ps1 @gradle @axmol-cmdline
@@ -66,11 +66,11 @@ agp=8.7.3
 
 # The android build-tools, @axmol-cmdline @gradle
 # as stable as possible, match with agp,android-studio
-build-tools=34.0.0
+buildtools=34.0.0
 
-# The android-studio, @hint
+# The android studio, @hint
 # as latest as possible, but match with agp, build-tools
-android-studio=2024.2.1+
+androidstudio=2024.2.1+
 
 # The android dependencies managed by @gradle 
 appcompat=1.7.0

--- a/1k/manifest.ps1
+++ b/1k/manifest.ps1
@@ -19,11 +19,6 @@ $manifest['cmake'] = $build_profiles['cmake']
 $manifest['vs'] = $build_profiles['vs']
 $manifest['llvm'] = $build_profiles['llvm']
 
-# android sdk tools
-$android_sdk_tools['build-tools'] = $build_profiles['build-tools']
-$android_sdk_tools['platforms'] = $build_profiles['target_sdk']
-$android_sdk_tools['cmdline-tools'] = $build_profiles['cmdline-tools']
-
 $Global:build_profiles = $build_profiles
 
 $Global:download_path = $1k.realpath("$PSScriptRoot/../cache")

--- a/core/platform/android/libaxmol/axmol.gradle
+++ b/core/platform/android/libaxmol/axmol.gradle
@@ -219,6 +219,9 @@ class AxmolUtils {
     }
 
     private static String translateNdkVer(ndkVer) {
+        if (ndkVer.endsWith('+')) {
+            ndkVer = ndkVer.substring(0, ndkVer.length() - 1);
+        }
         def ndkMajor = ndkVer.replaceAll('[^0-9]', '')
         def ndkMinorOff = ndkMajor.length() + 1
 

--- a/tools/cmdline/axmol
+++ b/tools/cmdline/axmol
@@ -1,13 +1,11 @@
 #!/bin/bash -l
 
-AXMOL_CONSOLE_BIN_DIRECTORY=$(dirname "$0")
-AXMOL_CONSOLE_BIN_DIRECTORY=$(cd "$AXMOL_CONSOLE_BIN_DIRECTORY" && pwd -P)
-
+SCRIPT_ROOT=$(dirname "$0")
+SCRIPT_ROOT=$(cd "$SCRIPT_ROOT" && pwd -P)
 if hash pwsh 2>/dev/null; then
     POWERSHELL=pwsh
 else
     echo "PowerShell 7+ required."
     exit 1
 fi
-
-$POWERSHELL "$AXMOL_CONSOLE_BIN_DIRECTORY/axmol.ps1" "$@"
+$POWERSHELL "$SCRIPT_ROOT/axmol.ps1" "$@"

--- a/tools/cmdline/build.ps1
+++ b/tools/cmdline/build.ps1
@@ -99,6 +99,10 @@ $Global:is_axmol_engine = ($source_proj_dir -eq $AX_ROOT)
 $Global:is_axmol_app = !!(Get-ChildItem (Join-Path $source_proj_dir '.axproj*'))
 $is_android = $options.p -eq 'android'
 
+if($Global:is_axmol_app) {
+    $Global:__1k_user_profile = Join-Path $source_proj_dir '.axproj'
+}
+
 # start construct full cmd line
 $1k_args = @()
 

